src/worker.jl

```diff
@@ -244,13 +244,14 @@ end
 
 function _augment_subproblems(record::WorkerRecord,
                               r::AbstractPenaltyParameter
-                              )::Nothing
+                              )
 
+    penalty_map = Dict{ScenarioID,Dict{VariableID,Real}}()
     for (scen, sub) in record.subproblems
-        add_ph_objective_terms(sub.problem,
+        penalty_map[scen] = add_ph_objective_terms(sub.problem,
```

Comment:
Instead of collecting these into a dictionary and returning that, you want to create a new `Message` subtype (see message.jl) then send that back to the master process using the `output::RemoteChannel` found in the calling function `process_message(msg::Initialize,...)`. That is, you'll need to add it to the function signature for `_augment_subproblems`.  An example can be found in `_initial_solve` in worker.jl.

We'll need to add code on the master side as well.  That will be in setup.jl.  The functions `_initialize_subproblem` and `_set_initial_values` will require changes and we'll need a new function that will process your new message type from all the workers and update the penalty parameter in the `PHData` struct.  The `initialize` function in setup.jl is the calling function for these.  Try to implement it and when you run into trouble (this is one of the most complex areas of the code!), we can talk about it. Initialization is the only time that messages of different types show up in the master remote channel so watch out for that.  The master side access to the remote channel is managed through the `WorkerInf` struct.  The definition and the needed functions are found in worker_management.jl.